core/src/test/scala/io/chrisdavenport/shellfish/FileOsSpec.scala

```diff
@@ -80,7 +80,8 @@ object FileOsSpec extends SimpleIOSuite with Checkers {
           sizeBefore <- path.readLines.map(_.size)
           _          <- path.appendLine("Im a last line!")
           sizeAfter  <- path.readLines.map(_.size)
-        } yield expect(sizeBefore + 1 == sizeAfter)
+        } yield expect(sizeBefore + 2 == sizeAfter)
+        // That is, one new line of the appendLine and, one newline character of the writeLines method
```

Comment:
If you've added the `dropLastIf(_.isEmpty)` how come you need this modification here?

Reply:
Before the change:
```scala
val aList = List("a", "b")

for 
  _          <- path.writeLines(contentsList)
  sizeBefore <- path.readLines.map(_.size) // sizeBefore is 3, as writeLines adds a newline at the end
  _          <- path.appendLine("Im a last line!") // Adds another line
  sizeAfter  <- path.readLines.map(_.size) // Now sizeAfter is 4
yield expect(sizeBefore + 1 == sizeAfter) // so this condition is true
```

Now
```scala
val aList = List("a", "b")

for 
  _          <- path.writeLines(contentsList)
  sizeBefore <- path.readLines.map(_.size) // sizeBefore is 2, as readLines now ignores all empty characters before the last newline
  _          <- path.appendLine("Im a last line!") // Adds another line
  sizeAfter  <- path.readLines.map(_.size) // Now sizeAfter is 4, as we didnt get rid of the newline added by writeLines
yield expect(sizeBefore + 1 == sizeAfter) // so this condition is not true anymore, as sizeBefore is 2 and sizeAfter is 4
```